stl/inc/algorithm

```diff
@@ -4745,7 +4745,8 @@ _CONSTEXPR20 bool next_permutation(_BidIt _First, _BidIt _Last, _Pr _Pred) {
 
     for (;;) { // find rightmost element smaller than successor
         auto _UNext1 = _UNext;
-        if (_DEBUG_LT_PRED(_Pred, *--_UNext, *_UNext1)) { // swap with rightmost element that's smaller, flip suffix
+        if (_DEBUG_LT_PRED(
+                _Pred, *_Prev_iter(_UNext), *_UNext1)) { // swap with rightmost element that's smaller, flip suffix

```

Comment:
This would format better if you insert a newline after the { to put the comment on the following line. (Ditto below)

Reply:
Oh okay let me do this then